ui/src/components/common/NotificationCenter.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+
+import Page from '../common/Page'
+import PageSection from './PageSection'
+import { useRelationshipRequestNotifications } from '../common/RelationshipRequestNotifications'
+import { useCCPCustomerNotifications } from '../CCP/CCPCustomerNotifications'
+import { useBrokerCustomerInviteNotifications } from '../Investor/BrokerCustomerInviteNotifications'
+import { useCCPCustomerInviteNotifications } from '../Investor/CCPInviteNotifications'
+import { useDismissibleNotifications } from '../common/DismissibleNotifications'
+import { useExchangeInviteNotifications } from '../Investor/ExchangeInviteNotifications'
+
+import { NotificationCenterIcon, ArrowLeftIcon } from '../../icons/Icons'
+
+type Props = {
+  sideNav: React.ReactElement;
+  onLogout: () => void;
+  showNotificationAlert?: boolean;
+  handleNotificationAlert?: () => void;
+}
+
+export const useAllNotifications = () => {
+  const notifications = [
+    ...useRelationshipRequestNotifications(),
+    ...useCCPCustomerNotifications(),
+    ...useCCPCustomerInviteNotifications(),
+    ...useBrokerCustomerInviteNotifications(),
+    ...useExchangeInviteNotifications(),
+    ...useDismissibleNotifications()
+  ];
+
+  return notifications;
+}
+
+const NotificationCenter: React.FC<Props> = ({ sideNav, onLogout, showNotificationAlert, handleNotificationAlert }) => {
+  const history = useHistory();
+  const notifications = useAllNotifications();
+  const [prevLocation, setPrevLocation] = useState<string>();
+  const [prevPath, setPrevPath] = useState<string>();
+
+  const historyState = history.location.state as string;
+
+  useEffect(() => {
+    const tempLocation = historyState?.split('/');
+    const tempPath = historyState;
+    if (historyState) {
+      localStorage.setItem('prevLocation', tempLocation[tempLocation.length - 1]);
```

Comment:
These would be better as `sessionStorage` (which stores per tab) instead of `localStorage` - with this if you have two tabs open on the notification center and refresh the first tab, it will have the back button use the route for the most recent tab instead of its own route.

Reply:
Ok! good to know :) I'll change it. It's the first time i hear of it...

Reply:
yup, I only learned about it last week fixing a similar issue. very handy info 😃